src/compiler/types.ts

```diff
@@ -415,7 +415,7 @@ namespace ts {
         BlockScoped = Let | Const,
 
         ReachabilityCheckFlags = HasImplicitReturn | HasExplicitReturn,
-        EmitHelperFlags = HasClassExtends | HasDecorators | HasParamDecorators | HasAsyncFunctions,
+        EmitHelperFlags = HasClassExtends | HasDecorators | HasParamDecorators | HasAsyncFunctions | HasJsxSpreadAttribute,

```

Comment:
Also, we need to consolidate the node flag values now that I've extracted the modifiers out of `NodeFlags`, that should help us avoid the SMI overflow on 32-bit node.